src/ua.c

```diff
@@ -493,23 +494,31 @@ static bool require_handler(const struct sip_hdr *hdr,
 }
 
 
-static int sdp_af_hint(struct mbuf *mb)
+static int sdp_originator(struct mbuf *mb, int *af, struct sa *sa)
 {
-	struct pl af;
+	struct pl pl1, pl2;
+	char *addr;
 	int err;
 
+	*af = AF_UNSPEC;
 	err = re_regex((char *)mbuf_buf(mb), mbuf_get_left(mb),
-		       "IN IP[46]+", &af);
+		       "o=[^ ]* [^ ]* [^ ]* IN IP[46]+ [^ \r\n]+",
+		       NULL, NULL, NULL, &pl1, &pl2);
 	if (err)
-		return AF_UNSPEC;
+		return EINVAL;
 
-	switch (af.p[0]) {
+	switch (pl1.p[0]) {
 
-	case '4': return AF_INET;
-	case '6': return AF_INET6;
+	case '4': *af = AF_INET;
+		  break;
+	case '6': *af = AF_INET6;
+		  break;
 	}
 
-	return AF_UNSPEC;
+	pl_strdup(&addr, &pl2);
+	err = sa_set_str(sa, addr, 5060);
```

Comment:
port 5060 is hardcoded here. The different SIP transports have different default ports,
e.g. 5061 for tls

Reply:
This is only a dummy port that makes `udp_connect()` work for OSX. Has nothing to do with SIP. It must not be zero for OSX.
Added:
```c
/* OSX/iOS needs a dummy port number for udp_connect() */
```

Reply:
what if the routing table has different rules for different ports ?

the port here should be the same as is used for the real traffic.

Reply:
Like here? https://www.system-rescue.org/networking/Destination-port-routing/
I think we should test if it is even discoverable with `net_dst_source_addr_get` (since it is a udp based dummy routing check.).
Maybe for special and not detectable cases its better to use `net_interface IP` in config or a account option?

Reply:
In this case it is the IP for the RTP stream. As we know, we have only one `laddr` for SDP. So I suggest to select `laddr` from which we have a route to destination IP:port where
- IP is taken from (c) line and
- the port number is taken from (m) **audio** line

Reply:
Added a commit.